src/app/sessions/page.tsx

```diff
@@ -19,18 +20,32 @@ import { Heading } from "@/components/Heading";
 import { MEMBER_MAX_SESSIONS, NON_MEMBER_MAX_SESSIONS } from "@/lib/constants";
 import { useCartStore } from "@/stores/useCartStore";
 
-const remainingSessions = 11;
-const isMember = true;
-const firstName = "David";
-
-const maxSessions = isMember ? MEMBER_MAX_SESSIONS : NON_MEMBER_MAX_SESSIONS;
+//const remainingSessions = 11;
+//const isMember = true;
+//const firstName = "David";
 
 export default function SelectSessionPage() {
   const { push } = useRouter();
 
   const sessionsSelected = useCartStore((state) => state.cart.length);
 
   const [shake, setShake] = useState(false);
+  const [firstName, setFirstName] = useState(null);
+  const [isMember, setIsMember] = useState(false);
+  const [remainingSessions, setRemainingSessions] = useState(false);
+  const maxSessions = isMember ? MEMBER_MAX_SESSIONS : NON_MEMBER_MAX_SESSIONS;
+
+  function getUserInfo(id: string) {
+    axios.get(`http://localhost:3000/api/user/${id}`).then((response) => {
+      const data = response.data;
+      setFirstName(data.firstName);
+      setIsMember(data.member);
+      setRemainingSessions(data.remainingSessions);
+      console.log(data);
+    });
+  }
+
+  getUserInfo("123");
```

Comment:
this is highly inefficient as this will make an api call every time the page is re-rendered. See the console output to see how many calls this is making to the database.

For our purposes, if we want to get some data we have a couple approaches:
- use react query (see the useCurrentGameSessions on how you can implement this), this allows us to handle async state through props such as isLoading, isSuccess, etc. 
- fetch server side, use server side components to fetch data on the server, and the pass the data down through props. If you choose to do it this way you can omit the awkward loading state where there is no information on the page.
    - This is relative easy to set up too.